### ISSUE TITLE
Capture links in components such as LinkCard or CardGrid

### ISSUE THREAD

**Issue by jbergstroem** [HiDeoo/starlight-links-validator#50]
### Describe the bug

When referencing pages in a CardGrid or LinkCard or your own custom component, the link checker does not check for consistency. It would be great if this was caught!

### To Reproduce

Example via cardgrid:
```astro
<CardGrid>
  <Card title="example" icon="rocket">
    - [link a](/link-a)
  </Card>
</CardGrid>
```

..if `/link-a` does not exist, there no error caught.

Same through the built-in component `CardGrid`:

```astro
<CardGrid>
  <LinkCard title="Link a" href="/link-a" />
</CardGrid>
```

### Expected behavior

The links validator would find these references and error since the destination does not exist

### How often does this bug happen?

Every time

### System Info

- starlight-links-validator: v0.10.1
- @astrojs/starlight: 0.25.4
- astro 4.14.5
